src/compiler/scanner.ts

```diff
@@ -613,13 +617,23 @@ namespace ts {
                 case CharacterCodes.greaterThan:
                     if (isConflictMarkerTrivia(text, pos)) {
                         pos = scanConflictMarkerTrivia(text, pos);
+                        canConsumeStar = false;
                         continue;
                     }
                     break;
 
                 case CharacterCodes.hash:
                     if (pos === 0 && isShebangTrivia(text, pos)) {
                         pos = scanShebangTrivia(text, pos);
+                        canConsumeStar = false;
+                        continue;
+                    }
+                    break;
+
+                case CharacterCodes.asterisk:
+                    if (canConsumeStar) {
+                        pos++;

```

Comment:
Dont you need to set canConsumeStar to false in the default case as well?

Reply:
I don’t think so—the default case is either whitespace (which doesn’t change whether we can consume a leading `*`) or it breaks and returns.